lib/metasploit/framework/login_scanner/base.rb

```diff
@@ -251,8 +251,13 @@ def scan!
                   break if total_error_count >= 10
                 end
               end
-            rescue => e
-              elog('Attempt may not yield a result', error: e)
+            rescue Rex::ConnectionRefused, Rex::SocketError, Rex::ConnectionError, Rex::ConnectionProxyError, Rex::StreamClosedError, Rex::TimeoutError, ::Timeout::Error, Errno::ETIMEDOUT, Errno::ENOTCONN, Errno::ECONNRESET => e
+              framework_module.print_warning("Error, scan may not produce results: #{e.message}") if framework_module
```

Comment:
It would be nice if this logged out the credential details that caused the issue - otherwise it doesn't align with the current output generated by the logging scanner

Reply:
```diff
diff --git a/lib/metasploit/framework/login_scanner/base.rb b/lib/metasploit/framework/login_scanner/base.rb
index 6b1ba2f820..6d0e5e9c85 100644
--- a/lib/metasploit/framework/login_scanner/base.rb
+++ b/lib/metasploit/framework/login_scanner/base.rb
@@ -251,8 +251,11 @@ module Metasploit
                   break if total_error_count >= 10
                 end
               end
-            rescue Rex::ConnectionRefused, Rex::SocketError, Rex::ConnectionError, Rex::ConnectionProxyError, Rex::StreamClosedError, Rex::TimeoutError, ::Timeout::Error, Errno::ETIMEDOUT, Errno::ENOTCONN, Errno::ECONNRESET => e
-              framework_module.print_warning("Error, scan may not produce results: #{e.message}") if framework_module
+            rescue => e
+              if framework_module
+                prefix = framework_module.respond_to?(:peer) ? "#{framework_module.peer} - LOGIN FAILED:" : "LOGIN FAILED:"
+                framework_module.print_warning("#{prefix} #{credential.to_h} - Unhandled error - scan may not produce correct results: #{e.message} - #{e.backtrace}")
+              end
               elog("Scan Error: #{e.message}", error: e)
               consecutive_error_count += 1
               total_error_count += 1

```

It's a bit verbose; but at least we'd be able to track down any issue that gets reported pretty fast

```
msf6 auxiliary(scanner/redis/redis_login) > run rhost=127.0.0.1 username=admin password=foo rport=6379

[!] 127.0.0.1:6379        - 127.0.0.1:6379        - LOGIN FAILED: {:private_data=>"foo", :private_type=>:password, :username=>nil, :realm_key=>nil, :realm_value=>nil} - Unhandled error - scan may not produce correct results: undefined method `strip' for nil:NilClass - ["/Users/user/Documents/code/metasploit-framework/modules/auxiliary/scanner/redis/redis_login.rb:110:in `block in run_host'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/login_scanner/base.rb:234:in `block in scan!'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/login_scanner/base.rb:179:in `block in each_credential'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/credential_collection.rb:94:in `block in each_filtered'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/credential_collection.rb:111:in `each_unfiltered'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/credential_collection.rb:91:in `each_filtered'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/login_scanner/base.rb:141:in `each_credential'", "/Users/user/Documents/code/metasploit-framework/lib/metasploit/framework/login_scanner/base.rb:205:in `scan!'", "/Users/user/Documents/code/metasploit-framework/modules/auxiliary/scanner/redis/redis_login.rb:83:in `run_host'", "/Users/user/Documents/code/metasploit-framework/lib/msf/core/auxiliary/scanner.rb:128:in `block (2 levels) in run'", "/Users/user/Documents/code/metasploit-framework/lib/msf/core/thread_manager.rb:105:in `block in spawn'"]
```